ntoskrnl/ex/time.c

```diff
@@ -334,8 +357,25 @@ ExpSetTimeZoneInformation(PRTL_TIME_ZONE_INFORMATION TimeZoneInformation)
 {
     LARGE_INTEGER LocalTime, SystemTime, OldTime;
     TIME_FIELDS TimeFields;
+    NTSTATUS Status = STATUS_UNSUCCESSFUL; // Allow easy failure return change
     DPRINT("ExpSetTimeZoneInformation() called\n");
 
+    /* Get the Shared User Data System Time into the local SystemTime */
+    do
+    {
+        SystemTime.HighPart = SharedUserData->SystemTime.High1Time;
+        SystemTime.LowPart = SharedUserData->SystemTime.LowPart;
+    }
+    while (SystemTime.HighPart != SharedUserData->SystemTime.High2Time);
+
+    /* Set the Global Data for ExpTimeZoneBias and ExpTimeZoneId */
+    if (!ExpGetTimeZoneId(&SystemTime, &ExpTimeZoneId))
+    {
+        DPRINT1("ExpSetTimeZoneInformation() failed (Status 0x%08lx)\n", Status);
+        return Status;
```

Comment:
```suggestion
        DPRINT1("ExpGetTimeZoneId() failed\n");
        return STATUS_UNSUCCESSFUL;
```
Not sure the dprint1 is that useful though.... (it already emits dprints when the functions it calls fail)

Reply:
Changed. Thanks.